packages/bruno-app/src/components/RequestPane/Auth/OAuth2/CredentialsPreview/index.js

```diff
@@ -0,0 +1,95 @@
+import React, { useEffect, useState } from 'react';
+import { clearOauth2Cache, readOauth2CachedCredentials } from 'utils/network';
+import { sendRequest } from 'providers/ReduxStore/slices/collections/actions';
+import toast from 'react-hot-toast';
+import { useDispatch } from 'react-redux';
+
+const CredentialsPreview = ({ item, collection }) => {
+  const oauth2CredentialsAreaRef = React.createRef();
+  const [oauth2Credentials, setOauth2Credentials] = useState({});
+  const [focusedField, setFocusedField] = useState(null);
+
+  const dispatch = useDispatch();
+
+  useEffect(() => {
+    oauth2CredentialsAreaRef.current.value = oauth2Credentials;
+    readOauth2CachedCredentials(collection.uid).then((credentials) => setOauth2Credentials(credentials));
+  }, [oauth2CredentialsAreaRef]);
+
+  const handleRun = async () => {
+    dispatch(sendRequest(item, collection.uid));
+  };
+
+  const handleClearCache = (e) => {
+    clearOauth2Cache(collection?.uid)
+      .then(() => {
+        readOauth2CachedCredentials(collection.uid).then((credentials) => {
+          setOauth2Credentials(credentials);
+          toast.success('cleared cache successfully');
+        });
+      })
+      .catch((err) => {
+        toast.error(err.message);
+      });
+  };
+
+  const handleFocus = (field) => {
+    setFocusedField(field);
+  };
+
+  const handleBlur = () => {
+    setFocusedField(null);
+  };
+
+  const sortedFields = () => {
+    const tokens = {};
+    const extras = {};
+    Object.entries(oauth2Credentials).forEach(([key, value]) => {
+      if (key.endsWith('_token')) {
+        tokens[key] = value;
+      } else {
+        extras[key] = value;
+      }
+    });
+    return { ...tokens, ...extras };
+  };
+
+  return (
+    <div className="flex flex-col w-full gap-1 mt-4">
+      <div className="flex flex-row gap-4">
+        <button onClick={handleRun} className="submit btn btn-sm btn-secondary w-fit">
+          Get Access Token
+        </button>
+        <button onClick={handleClearCache} className="submit btn btn-sm btn-secondary w-fit">
+          Clear Cache
+        </button>
+      </div>
+      <div ref={oauth2CredentialsAreaRef}>
+        <label className="flex flex-row w-full mt-2 gap-2">Cached OAuth2 Credentials:</label>
+        {Object.entries(sortedFields()).map(([field, value]) => (
+          <div key={field}>
+            <label className="text-xs">{field}</label>
+            {focusedField === field || !field.endsWith('_token') ? (
+              <textarea className="w-full h-24 p-2 text-xs border rounded" value={value} onBlur={handleBlur} readOnly />
```

Comment:
I noticed that when blurred, you can still scroll. Slight nitpick but if we could disable that until you un-blur it, that would be ideal

Reply:
I definitely agree, code duplication like that increases the mental load for developers, and also means that making a UI change is twice (or thrice!) the work. We should have shared components as much as possible

Reply:
After playing around with UI, I decided to move away from blur effect - towards the solution where entire credential section is collapsed on demand. Helps to declutter the screen without the need to lose access token in the process.

Reply:
I like it! One small thing is maybe instead of the `{+}` or `{-}`, using arrows similar to this:

<details><summary>Cached OAuth2 Credentials</summary>
<p>

Auth content goes here

</p>
</details>

Reply:
TIL: `<details>` and `<summary>` tags.